packages/website/blog/2024-03-01-parser-options-project-service.md

```diff
@@ -0,0 +1,449 @@
+
+<!--tabs-->
+
+### Flat ESLint Config
+
+```js title="eslint.config.js"
+export default tseslint.flatConfig({
+  // ...
+  languageOptions: {
+    parser: tseslint.parser,
+    parserOptions: {
+      projectService: {
+        additionalFiles: ['./*'],
+      },
+      tsconfigRootDir: __dirname,
+    },
+  },
+  // ...
+});
+```
+
+### Legacy ESLint Config
+
+```js title=".eslintrc.cjs"
+module.exports = {
+  // ...
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    projectService: {
+      additionalFiles: ['./*'],
+    },
+    tsconfigRootDir: __dirname,
+  },
+  // ...
+};
+```
+
+<!--/tabs-->
+
+:::tip
+This means you should be able to remove any lint-only `tsconfig.eslint.json` files!
+🥳
+:::
+
+Note that `projectService.additionalFiles` should only include files that aren't included in the project's `tsconfig.json`.
+Creating fallback type information for additional files is slower than including those files in the `tsconfig.json`.
+If a file is included in both the `tsconfig.json` and `projectService.additionalFiles`, typescript-eslint will report a parsing error:
+
+```plaintext
+${filePath} was included by projectService.additionalFiles but also was found in the project service. Consider removing it from projectService.additionalFiles.
+```
+
+See [feat(typescript-estree): add allowDefaultProjectForFiles project service allowlist option](https://github.com/typescript-eslint/typescript-eslint/pull/7752) for more details on the API and [chore: add performance package with a project service hyperfine comparison (#7870)](https://github.com/typescript-eslint/typescript-eslint/pull/7870) for more details on performance.
+
+### New Starter Configs
+
+The following ESLint configs are starting recommendations for typed linting using the project service:
+
+<!--tabs-->
+
+## Flat ESLint Config
+
+```js title="eslint.config.js"
+import tseslint from '@typescript-eslint/core';
+import js from '@eslint/js';
+
+export default tseslint.flatConfig({
+  extends: [js.configs.recommended, tseslint.configs.recommendedTypeChecked],
+  plugins: {
+    '@typescript-eslint': tseslint.plugin,
+  },
+  languageOptions: {
+    parser: tseslint.parser,
+    parserOptions: {
+      projectService: {
+        additionalFiles: ['./*'],
+      },
+      tsconfigRootDir: __dirname,
+    },
+  },
+});
+```
+
+## Legacy ESLint Config
+
+```js title=".eslintrc.cjs"
+module.exports = {
+  extends: [
+    'eslint:recommended',
+    'plugin:@typescript-eslint/recommended-type-checked',
+  ],
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    projectService: {
+      additionalFiles: ['./*'],
+    },
+    tsconfigRootDir: __dirname,
+  },
+  plugins: ['@typescript-eslint'],
+  root: true,
+};
+```
+
+<!--/tabs-->
+
+See [typescript-eslint/examples](https://github.com/typescript-eslint/examples) for more references of how to configure linting in different projects.
+
+## Performance Comparisons
```

Comment:
Maybe also good to have - https://github.com/typescript-eslint/typescript-eslint/issues/7906 ...